operator/k8s_node.go

```diff
@@ -44,22 +44,30 @@ import (
 )
 
 func runNodeWatcher(nodeManager *allocator.NodeEventHandler) error {
```

Comment:
It's preferable to have 2 functions, one where it's explicitly that kvstore is going to be used and another where CRD is used. This way there's no need to have `if option.Config.KVStore != "" {` inside the function and would result in a cleaner code.

Reply:
I have refactored the function into two. 

I understand that the one with kvstore enabled would be used here https://github.com/cilium/cilium/blob/b667f010fb814df3e056f18c39933e02e4b93bfb/operator/main.go#L492

Where would be the watcher with disabled kvstore be used? Would adding an else condition to the parent if block make sense?

Reply:
>Would adding an else condition to the parent if block make sense?

yes and no. Yes that's the right location to call `runNodeWatcher` and no because we only need to run `runNodeWatcher` and if operatorOption.Config.CNPNodeStatusGCInterval != 0. So we should probably do an early return in this function if operatorOption.Config.CNPNodeStatusGCInterval != 0.